libsigner/src/http.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::io;
+use std::io::{Read, Write};
+
+use std::net::SocketAddr;
+
+use stacks_common::codec::MAX_MESSAGE_LEN;
+use stacks_common::deps_common::httparse;
+use stacks_common::util::chunked_encoding::*;
+
+use crate::error::EventError;
+use crate::error::RPCError;
+
+pub const MAX_HTTP_HEADERS: usize = 32;
+pub const MAX_HTTP_HEADER_LEN: usize = 4096;
+
+/// Decoding of the relevant parts of a signer-directed HTTP request from the Stacks node
+#[derive(Debug)]
+pub struct SignerHttpRequest {
```

Comment:
Since we're using `tiny_http` now, probably all of this can go away.

Reply:
This happened in the first `Signer` PR:

https://github.com/stacks-network/stacks-blockchain/blob/feat/stacks-signer/libsigner/Cargo.toml#L29
https://github.com/stacks-network/stacks-blockchain/blob/feat/stacks-signer/libsigner/src/events.rs#L33
https://github.com/stacks-network/stacks-blockchain/blob/feat/stacks-signer/libsigner/src/events.rs#L201

We're currently using it as an `HTTP` server for `StackerDBEventReceiver`.  So all of the server side `HTTP` parsing can go away in `libsigner`, since `tiny_http` comes with its own parser and request/response objects, along with all the tests.

In a later PR we can explore getting rid of `async_h1`, which is used for the client side of the stacks event observer code.

Reply:
So AFAICT all of the `libsigner` custom `HTTP` code is now dead.